src/object_.d

```diff
@@ -169,8 +169,11 @@ bool opEquals(Object lhs, Object rhs)
     if (lhs is null || rhs is null) return false;
 
     // If same exact type => one call to method opEquals
-    if (typeid(lhs) is typeid(rhs) || typeid(lhs).opEquals(typeid(rhs)))
+    if (!__ctfe && // CTFE doesn't like comparing typeid
```

Comment:
Sorry my description was a little wrong. Originally there was only one equality comparison for TypeInfo objects `typeid(lhs) == typeid(rhs) `. It was changed to current code for the efficiency.

https://issues.dlang.org/show_bug.cgi?id=3917
https://github.com/D-Programming-Language/druntime/commit/8409adbbb6bde641bd2b852637016068d22c6756

But my proposal is still correct. During CTFE, all TypeInfo objects are purely singletons. So just only using identity comparison is enough.